src/components/lib/Tooltip.js

```diff
@@ -0,0 +1,59 @@
+import React, { useEffect, useRef, useState } from "react"
+import classNames from "classnames"
+import PropTypes from "prop-types"
+
+const Tooltip = ({
+  triggerElement = <button>Help</button>,
+  children,
+  revealOnHover = true,
+  className,
+}) => {
+  const [showContent, setShowContent] = useState(false)
+  const toggleContent = () => {
+    setShowContent(!showContent)
+  }
+
+  const tooltipRef = useRef(null)
+  useEffect(() => {
+    const clickOutside = (e) => {
+      if (tooltipRef.current && !tooltipRef.current.contains(e.target)) {
+        setShowContent(false)
+      }
+    }
+
+    // Only add the listener if the dropdown is open, and make sure to clean
+    // it up once it's closed
+    if (showContent) {
+      document.addEventListener("click", clickOutside)
+    } else {
+      document.removeEventListener("click", clickOutside)
+    }
+  }, [tooltipRef, showContent])
+
+  return (
+    <div
+      ref={tooltipRef}
+      className={classNames("tooltip", className, { hoverable: revealOnHover })}
+    >
+      {React.cloneElement(triggerElement, {
```

Comment:
Why clone the element here?

Reply:
To attach the listener? Hmm, maybe I can try to just attach it normally.

Reply:
(It won't let me do this `<TriggerElement onClick={toggleContent} />` ☹️ )

Reply:
Ahhh I see. I think this is fine, mostly it was curiosity on my part.